extensions/sql-bindings/src/test/common/azureFunctionsUtils.test.ts

```diff
@@ -0,0 +1,103 @@
+
+describe('Tests to verify Azure Functions Utils functions', function (): void {
+
+	it('Should get local.settings.json', async () => {
+		sinon.stub(fs, 'existsSync').withArgs(localSettingsPath).returns(true);
+		sinon.stub(fs, 'readFileSync').withArgs(localSettingsPath).returns(
+			`{"IsEncrypted": false,
+			"Values": {"test1": "test1", "test2": "test2", "test3":"test3"}}`
+		).toString();
+		let settings = await azureFunctionsUtils.getLocalSettingsJson(localSettingsPath);
+		should(settings.IsEncrypted).equals(false);
+		should(Object.keys(settings.Values!).length).equals(3);
+	});
+
+	it('Should show error message if local.settings.json throws', async () => {
+		let errorMsg = 'Error parsing local.settings.json';
+		sinon.stub(fs, 'existsSync').withArgs(localSettingsPath).returns(true);
+		sinon.stub(fs, 'readFileSync').withArgs(localSettingsPath).returns(
+			`{"IsEncrypted": false,
+			"Values": {"test1": "test1", "test2": "test2", "test3":"test3"}}`
+		).toString();
+		const getLocalSettingsSpy = sinon.spy(azureFunctionsUtils, 'getLocalSettingsJson');
+		sinon.stub(JSON, 'parse').withArgs(sinon.match.any).throws(new Error(errorMsg));
+		try {
+			await getLocalSettingsSpy(localSettingsPath);
+		} catch {
+			should(getLocalSettingsSpy.threw());
+		}
+	});
+
+	it('Should set local.settings.json with new value', async () => {
+		sinon.stub(fs, 'existsSync').withArgs(localSettingsPath).returns(true);
+		sinon.stub(fs, 'readFileSync').withArgs(localSettingsPath).returns(
+			`{"IsEncrypted": false,
+			"Values": {"test1": "test1", "test2": "test2", "test3":"test3"}}`
+		).toString();
+		let settings = await azureFunctionsUtils.getLocalSettingsJson(localSettingsPath);
+		// originally should have 3 settings
+		should(Object.keys(settings.Values!).length).equals(3);
+
+		let writeFileStub = sinon.stub(fs.promises, 'writeFile');
+		await azureFunctionsUtils.setLocalAppSetting(path.dirname(localSettingsPath), 'test4', 'test4');
+		should(writeFileStub.calledWithExactly(localSettingsPath, '{\n  "IsEncrypted": false,\n  "Values": {\n    "test1": "test1",\n    "test2": "test2",\n    "test3": "test3",\n    "test4": "test4"\n  }\n}')).equals(true);
+	});
+
+	it('Should not overwrite setting if value already exists in local.settings.json', async () => {
+		sinon.stub(fs, 'existsSync').withArgs(localSettingsPath).returns(true);
+		sinon.stub(fs, 'readFileSync').withArgs(localSettingsPath).returns(
+			`{"IsEncrypted": false,
+			"Values": {"test1": "test1", "test2": "test2", "test3":"test3"}}`
+		).toString();
+
+		let warningMsg = constants.settingAlreadyExists('test1');
+		let settings = await azureFunctionsUtils.getLocalSettingsJson(localSettingsPath);
+		// originally should have 3 settings
+		should(Object.keys(settings.Values!).length).equals(3);
+
+		const spy = sinon.stub(vscode.window, 'showWarningMessage').resolves(await Promise.resolve({ title: constants.settingAlreadyExists('test1') }));
```

Comment:
The `await Promise.resolve` should be unnecessary, you're already using the resolves function...